var/spack/repos/builtin/packages/slurm/package.py

```diff
@@ -179,8 +179,8 @@ class Slurm(AutotoolsPackage):
     @classmethod
     def determine_version(cls, exe):
         output = Executable(exe)("--version", output=str).rstrip()
-        match = re.search(r"slurm\s*([0-9.]+)", output)
-        return match.group(1) if match else None
+        match = re.search(r"slurm\s*([0-9.]+)|slurm-wlm\s*([0-9.]+)", output)
+        return (match.group(1) or match.group(2)) if match else None
```

Comment:
You can use a [non-capturing group](https://stackoverflow.com/questions/3512471/what-is-a-non-capturing-group-in-regular-expressions) (which is optional) to avoid the ambiguity in which group to use.
```suggestion
        match = re.search(r"slurm(?:-wlm)?\s*([0-9.]+)", output)
        return match.group(1) if match else None
```